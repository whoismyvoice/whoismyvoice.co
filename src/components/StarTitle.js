import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

import '../styles/TitleComponent.css';

export class StarTitle extends Component {
  static defaultProps = {
    templateData: {},
  };

  static propTypes = {
    templateData: PropTypes.shape({}),
    templateString: PropTypes.string.isRequired,
  };

  constructor(props) {
    super(props);
    this.template = _.template(props.templateString);
  }

  componentWillUpdate(nextProps, nextState, nextContext) {
    if (this.props.templateString !== nextProps.templateString) {
      this.template = _.template(nextProps.templateString);
    }
  }

  renderStars() {
    return (
      <div className="three-stars">
        <span>&#9733;</span>
        <span>&#9733;</span>
        <span>&#9733;</span>
      </div>
    );
  }

  render() {
    const { templateData } = this.props;
    const content = { __html: this.template(templateData) };
    return (
      <div className="title-component uppercase">
        {this.renderStars()}
        <div
          className="title-component__description"
          dangerouslySetInnerHTML={content}
        />
      </div>
    );
  }
}

export default StarTitle;
