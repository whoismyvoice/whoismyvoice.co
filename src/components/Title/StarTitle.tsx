import React, { FC } from 'react';

import '../../styles/TitleComponent.scss';

const Stars: FC<unknown> = () => (
  <div className="three-stars">
    <span>&#9733;</span>
    <span>&#9733;</span>
    <span>&#9733;</span>
  </div>
);

export const StarTitle: FC<unknown> = ({ children }) => (
  <div className="title-component uppercase">
    <Stars />
    <div className="title-component__description">{children}</div>
  </div>
);

export default StarTitle;
