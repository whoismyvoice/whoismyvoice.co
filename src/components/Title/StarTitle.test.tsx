import * as React from 'react';
import * as renderer from 'react-test-renderer';
import StarTitle from './StarTitle';

jest.mock('mixpanel-browser');

it('renders correctly no props', () => {
  const tree = renderer.create(<StarTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});
