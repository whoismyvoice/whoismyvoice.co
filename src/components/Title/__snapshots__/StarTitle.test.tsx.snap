// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly no props 1`] = `
<div
  className="title-component uppercase"
>
  <div
    className="three-stars"
  >
    <span>
      ★
    </span>
    <span>
      ★
    </span>
    <span>
      ★
    </span>
  </div>
  <div
    className="title-component__description"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
